src/types/discord.ts

```diff
@@ -363,3 +363,15 @@ export interface InviteDeleteEvent {
   /** the unique invite code */
   code: string;
 }
+
+/** https://discord.com/developers/docs/topics/gateway#update-status-gateway-status-update-structure */
+export interface GatewayStatusUpdatePayload {
+  /** unix time (in milliseconds) of when the client went idle, or null if the client is not idle */
+  since: number | null;
+  /** null, or the user's activities */
+  activities: Activity[] | null;
```

Comment:
```suggestion
  activities: Pick<ActivityPayload, "name" | "type" | "url">[] | null;
```
Bots are only allowed to use `name`, `type` and `url`. I am not sure if bots are allowed to have multiple activities.